File: Data Structures & Algorithms/duplicate-integer/submission-4.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        HashSet<int> numSet = new HashSet<int>();

        for(int i = 0; i<nums.Length; i++){
            numSet.Add(nums[i]);
        }

        return numSet.Count != nums.Length;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
from queue import PriorityQueue
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        pq = PriorityQueue()

        freq = {}
        result = []
        for num in nums:
            if num in freq:
                freq[num] += 1
            else:
                freq[num] = 1
        

        for key, values in freq.items():
            pq.put((-values, key))
        
        count = 0
        while count<k:
            priority, num = pq.get()
            result.append(num)
            count+=1
        
        return result
        
        